src/java.base/share/classes/sun/security/ssl/SSLConfiguration.java

```diff
@@ -403,10 +406,15 @@ boolean isAvailable(SSLExtension extension,
     void toggleClientMode() {
         this.isClientMode ^= true;
 
-        // reset the signature schemes
-        this.signatureSchemes = isClientMode ?
-                CustomizedClientSignatureSchemes.signatureSchemes :
-                CustomizedServerSignatureSchemes.signatureSchemes;
+        // Reset the signature schemes, if it was configured with SSLParameters.
+        if (Arrays.equals(signatureSchemes,
```

Comment:
The duals here does not mean it was configure with parameters, it only means it is currently the same configuration as the parameters, but it still could be a different source?

Reply:
You are right.  Toggle client mode is a pretty hard to implement method to me, especially considering the impact of various SSL parameters.  There are more than one method to set the parameters, as make it hard to know what an application really want to do.  Here is a best effort I can think of, but not really an accurate implementation.  Hopefully, the impact is minimal.